File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/mediator/ConcreteModel.java
package designPatterns.behavioral.mediator;

public class ConcreteModel extends Model {
  public ConcreteModel(Mediator mediator, String name) {
    super(mediator, name);
  }

  @Override
  public String send(String msg) {
    mediator.sendMessage(msg, this);
    return name + ": Sending Message = " + msg;
  }

  @Override
  public String receive(String msg) {
    return name + ": Received Message = " + msg;
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/mediator/ConcreteModel.kt
package behavioral.mediator

class ConcreteModel(mediator: Mediator, name: String) : Model(mediator, name) {
  override fun send(msg: String): String {
    mediator.sendMessage(msg, this)
    return "$name: Sending Message = $msg"
  }

  override fun receive(msg: String): String = "$name: Received Message = $msg"
}